src/main/scala/KindProjector.scala

```diff
@@ -200,19 +200,19 @@ class KindRewriter(plugin: Plugin, val global: Global)
 
       tree match {
         case Template(parents, self, body) =>
-          Template(parents.map(transform), self, body.map(transform))
+          treeCopy.Template(tree, parents.map(transform), self, body.map(transform))
```

Comment:
@non What is the purpose of this case? `super.transform(tree)` would perform:

```
treeCopy.Template(tree, transformTrees(parents), transformValDef(self), transformStats(body, tree.symbol))
```

Are you intentionally avoiding descent into the self type annotation? Or is the use of `atOwner` in `transformStats` problematic somehow?